### ISSUE TITLE
Crash upon process exit

### ISSUE THREAD

**Issue by rolfbjarne** [mono/mono#15518]
## Steps to Reproduce

Unfortunately I have very little information about this crash, it happened on a CI bot when building a sample (1 sample build out of almost 3700 sample builds; it's not related to that particular build).

This seems to be the most relevant part of the native stack trace:

```
=================================================================
	Native stacktrace:
=================================================================
	0x101cbf888 - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : mono_dump_native_crash_info
	0x101c5a105 - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : mono_handle_native_crash
	0x101cb9c46 - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : altstack_handle_and_restore
	0x101ec3c59 - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : monoeg_g_hash_table_lookup
	0x101bb9cc1 - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : mono_get_seq_points
	0x101c5738f - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : mono_walk_stack_full
	0x101c56ffc - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : mono_walk_stack_with_ctx
	0x101d97059 - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : mono_stack_walk
	0x101df6dcb - /Library/Frameworks/Mono.framework/Versions/6.0.0/bin/mono : mono_thread_interruption_checkpoint
```

in any case the complete native stack trace is available [here](https://gist.github.com/rolfbjarne/cfbd340f3c4323afc97d02b965d7bb09#file-build-log-L10095).

This seems to be when msbuild exits (because the last line before the crash is [Time Elapsed ...](https://gist.github.com/rolfbjarne/cfbd340f3c4323afc97d02b965d7bb09#file-build-log-L10085) which is usually the last line msbuild prints).

## On which platforms did you notice this

[x] macOS
[ ] Linux
[ ] Windows

**Version Used**:

```
$ mono --version
Mono JIT compiler version 5.12.0.301 (2018-02/4fe3280bba1 Fri Jul 20 08:25:42 EDT 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(3.6.0svn-mono-master/8b1520c8aae)
	GC:            sgen (concurrent by default)
``` 

Reference: https://dev.azure.com/xamarin/internal/_build/results?buildId=3772&view=logs

**Comment by rolfbjarne** [comment]
Happened again, but with line numbers now:

```
thread #6, name = 'RequestBuilder thread'
  frame #0: 0x00007fff6d77a356 libsystem_kernel.dylib`__wait4 + 10
  frame #1: 0x000000010d02cac0 mono`mono_dump_native_crash_info at mini-posix.c:1115:3 [opt]
  frame #2: 0x000000010d02c824 mono`mono_dump_native_crash_info(signal=<unavailable>, ctx=<unavailable>, info=<unavailable>) at mini-posix.c:1159 [opt]
  frame #3: 0x000000010cfc7105 mono`mono_handle_native_crash(signal="SIGSEGV", ctx=0x0000000000000000, info=0x0000000000000000) at mini-exceptions.c:3336:2 [opt]
  frame #4: 0x000000010d026c46 mono`altstack_handle_and_restore(ctx=0x000070000c228b80, obj=0x0000000000000000, stack_ovf=0) at exceptions-amd64.c:870:3 [opt]
  frame #5: 0x000000010d230c59 mono`monoeg_g_hash_table_lookup [inlined] monoeg_g_hash_table_lookup_extended(hash=<unavailable>) at ghashtable.c:332:37 [opt]
  frame #6: 0x000000010d230c21 mono`monoeg_g_hash_table_lookup(hash=<unavailable>, key=0x00007ffbc305cf00) at ghashtable.c:313 [opt]
  frame #7: 0x000000010cf26cc1 mono`mono_get_seq_points(domain=0x00007ffbc2c12430, method=0x00007ffbc305cf00) at seq-points.c:271:35 [opt]
  frame #8: 0x000000010cfc438f mono`mono_walk_stack_full(func=(mono`stack_walk_adapter at loader.c:2284), start_ctx=<unavailable>, domain=0x00007ffbc2c12430, jit_tls=0x00007ffb83810600, lmf=0x000070000c229648, unwind_options=<unavailable>, user_data=<unavailable>, crash_context=<unavailable>) at mini-exceptions.c:1342:19 [opt]
  frame #9: 0x000000010cfc3ffc mono`mono_walk_stack_with_ctx(func=(mono`stack_walk_adapter at loader.c:2284), start_ctx=0x000070000c2293c0, unwind_options=MONO_UNWIND_LOOKUP_ALL, user_data=<unavailable>) at mini-exceptions.c:1181:2 [opt]
  frame #10: 0x000000010d104059 mono`mono_stack_walk(func=<unavailable>, user_data=<unavailable>) at loader.c:2309:2 [opt]
  frame #11: 0x000000010d163dcb mono`mono_thread_interruption_checkpoint [inlined] is_running_protected_wrapper at threads.c:2852:2 [opt]
  frame #12: 0x000000010d163db4 mono`mono_thread_interruption_checkpoint [inlined] mono_thread_interruption_checkpoint_request(bypass_abort_protection=0) at threads.c:5072 [opt]
  frame #13: 0x000000010d163d74 mono`mono_thread_interruption_checkpoint at threads.c:5086 [opt]

```

[Full log](https://gist.github.com/rolfbjarne/9cb54943bfe9bc4f14ae6b01db518034#file-build-log-L4956-L4967)
Reference: https://dev.azure.com/xamarin/internal/_build/results?buildId=3820&view=logs

**Comment by lambdageek** [comment]
Looks like the same issue as #17737 which is fixed by #17738 (master) and #17741 (2019-10)
